### PR TITLE
[Y2K-942] Blocking on user profiles

### PR DESCRIPTION
**Pull request by jzila**
- [ ] Once #20866 is merged, switch the base branch of this PR to master.

### CODE REVIEW

**Comment by jzila** [comment]
cc @keybase/y2ksquad

File: go/libkb/user_card.go [hunk 1]
```diff
@@ -22,6 +22,10 @@ type card struct {
 	TeamShowcase         []keybase1.UserTeamShowcase `json:"team_showcase"`
 	RegisteredForAirdrop bool                        `json:"airdrop_registered"`
 	Blocked              bool                        `json:"blocked"`
+	UserBlocks           struct {
```
Comment (jakob223) [inline go/libkb/user_card.go]: feels weird to me that you're putting all of *my* user blocks in an `card` corresponding to someone else's profile.

File: shared/tracker2/bio/index.tsx [hunk 2]
```diff
@@ -177,6 +180,24 @@ const Bio = (p: Props) => (
...
+        </Kb.Text>
+        <Kb.Text type="BodySmallError" center={true} style={styles.text} selectable={true}>
+          {p.username} won’t be able to chat with you or add you to teams.
+        </Kb.Text>
+      </Kb.Text>
+    ) : (
+      p.hidFromFollowers && (
+        <Kb.Text type="BodySmallError" center={true} style={styles.blockedBackgroundText}>
+          <Kb.Text type="BodySmallError" center={true} style={styles.text} selectable={true}>
+            You hid them from your followers.
```
Comment (jakob223) [inline shared/tracker2/bio/index.tsx]: what if both?

File: go/libkb/user_card.go (hunk 1 above)
Comment (jzila) [reply go/libkb/user_card.go]: We aren't. This is just how the blocks pertaining to that user are returned by the server. Note that it's not an array. It's just the chat and follow bits for this target user.

File: shared/tracker2/bio/index.tsx (hunk 2 above)
Comment (jzila) [reply shared/tracker2/bio/index.tsx]: We show the blocked one, as Cecile responded to me on the design.

File: go/libkb/user_card.go (hunk 1 above)
Comment (jzila) [reply go/libkb/user_card.go]: You're right, these should be booleans.

Comment (jzila) [reply go/libkb/user_card.go]: I guess I was testing with the admin service instead of dev service. Oops!

**Pull request merged**